nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/kinesis/stream/record/KinesisRecordProcessorRaw.java

```diff
@@ -38,12 +39,16 @@ public KinesisRecordProcessorRaw(final ProcessSessionFactory sessionFactory, fin
     }
 
     @Override
-    void processRecord(final List<FlowFile> flowFiles, final Record kinesisRecord, final boolean lastRecord,
+    void processRecord(final List<FlowFile> flowFiles, final KinesisClientRecord kinesisRecord, final boolean lastRecord,
                        final ProcessSession session, final StopWatch stopWatch) {
-        final String partitionKey = kinesisRecord.getPartitionKey();
-        final String sequenceNumber = kinesisRecord.getSequenceNumber();
-        final Date approximateArrivalTimestamp = kinesisRecord.getApproximateArrivalTimestamp();
-        final byte[] data = kinesisRecord.getData() != null ? kinesisRecord.getData().array() : new byte[0];
+        final String partitionKey = kinesisRecord.partitionKey();
+        final String sequenceNumber = kinesisRecord.sequenceNumber();
+        final Instant approximateArrivalTimestamp = kinesisRecord.approximateArrivalTimestamp();
+        final ByteBuffer dataBuffer = kinesisRecord.data();
+        byte[] data = dataBuffer != null ? new byte[dataBuffer.remaining()] : new byte[0];
+        if (data != null) {
```

Comment:
```suggestion
        if (dataBuffer != null) {
```

`data` is never `null` here, but `dataBuffer` might be